File: backend/ULA-1/src/main/java/com/ula/models/TeachingTerm.java
package com.ula.models;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@Entity
@Table(name = "teaching_terms")
public class TeachingTerm extends BaseEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false, columnDefinition = "DATETIME")
	private Date startTime;

	@Column(nullable = false, columnDefinition = "DATETIME")
	private Date endTime;

	@ManyToOne
	private SubjectRealization subjectRealization;

	@OneToOne(mappedBy = "teachingTerm")
	private TeachingType teachingType;
}

File: frontend/projects/main-app/src/app/core/models/TeachingTerm.ts
export interface TeachingTerm {
    id: number;
    subjectRealizationId: number;
    type: {
        id: number;
        name: string;
    };
    teachingOutcome: {
        id: number;
        teachingTermId: number;
        description: string;
    };
    startTime: Date;
    endTime: Date;
}
